pkg/typesystem/resolver.go

```diff
@@ -45,37 +45,38 @@ func MemoizedTypesystemResolverFunc(datastore storage.AuthorizationModelReadBack
 			}
 		}
 
+		var v interface{}
+		var key string
 		if modelID == "" {
-			v, err, _ := lookupGroup.Do(fmt.Sprintf("FindLatestAuthorizationModelID:%s", storeID), func() (interface{}, error) {
-				return datastore.FindLatestAuthorizationModelID(ctx, storeID)
+			v, err, _ = lookupGroup.Do(fmt.Sprintf("FindLatestAuthorizationModel:%s", storeID), func() (interface{}, error) {
+				return datastore.FindLatestAuthorizationModel(ctx, storeID)
 			})
 			if err != nil {
 				if errors.Is(err, storage.ErrNotFound) {
 					return nil, ErrModelNotFound
 				}
 
-				return nil, fmt.Errorf("failed to FindLatestAuthorizationModelID: %w", err)
+				return nil, fmt.Errorf("failed to FindLatestAuthorizationModel: %w", err)
 			}
 
-			modelID = v.(string)
-		}
-
-		key := fmt.Sprintf("%s/%s", storeID, modelID)
+			key = fmt.Sprintf("%s/%s", storeID, v.(*openfgav1.AuthorizationModel).GetId())
```

Comment:
nit (non-blocking): it'd be nice to separate the type assertion from the getter.

```go
model := v.(*openfgav1.AuthorizationModel)

key = fmt.Sprintf("%s/%s", storeID, model.GetId())
```